tcms/telemetry/static/telemetry/js/testing/execution-dashboard.js

```diff
@@ -16,20 +16,37 @@ function preProcessData (data, callbackF) {
 
     jsonRPC('TestCase.filter', { pk__in: caseIds }, function (cases) {
         const testerPerCase = {}
+        const componentsPerCase = {}
         cases.forEach(function (element) {
             testerPerCase[element.id] = element.default_tester__username
+
+            jsonRPC('Component.filter', {cases: element.id}, function (components) {
```

Comment:
I think this can also be adjusted to filter all components, for all test cases with a single call (instead of as part of a loop). That should be much more efficient. See for example how the call to `TestRun.filter` is given multiple IDs as argument.